source/s3_request_messages.c

```diff
@@ -608,13 +670,48 @@ struct aws_input_stream *aws_s3_message_util_assign_body(
         goto error_clean_up;
     }
 
+    bool is_get = s_is_get_request(out_message);
```

Comment:
Will get request ever assign body? Should this just be an ASSERT?

Reply:
Yes, a default/single part get will be assigned a body.

Reply:
The get request should not have body. 
https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html Request Body section.

The default request does support the case for both body exist and not exist.

When there is no body, it should NOT call this function.

Reply:
updated default requests to only assign if not a get request. Though, it is kind of odd to only check for get requests, and no other request types. so idk if that check really makes sense, a default head would still have it's body assigned. A copy object request (which is a Put with no body) will have a body assigned.

Reply:
I have not seen any update

Reply:
Sorry didn't push, pushed now. Ran into errors moving checksum options from client settings to request settings. Testing infra really not built to support that.